tests/derive/groups.rs

```diff
@@ -120,3 +120,48 @@ fn helpful_panic_on_duplicate_groups() {
     use clap::CommandFactory;
     Opt::command().debug_assert();
 }
+
+#[test]
+fn required_group() {
+    #[derive(Parser, Debug)]
+    struct Opt {
+        #[command(flatten)]
+        source: Source,
+
+        #[arg(long)]
+        alt_source: String,
+    }
+
+    #[derive(clap::Args, Debug)]
+    #[group(required = true, multiple = false, conflicts_with = "alt_source")]
+    struct Source {
+        #[arg(long)]
+        path: Option<std::path::PathBuf>,
+        #[arg(long)]
+        git: Option<String>,
+    }
+
+    const OUTPUT: &str = "\
+error: the following required arguments were not provided:
+  --alt-source <ALT_SOURCE>
+  <--path <PATH>|--git <GIT>>
+
+Usage: prog --alt-source <ALT_SOURCE> <--path <PATH>|--git <GIT>>
+
+For more information, try '--help'.
+";
+    assert_output::<Opt>("prog", OUTPUT, true);
+
+    use clap::Args;
+    assert_eq!(Opt::group_id(), Some(clap::Id::from("Opt")));
+    assert_eq!(Source::group_id(), Some(clap::Id::from("Source")));
+    use clap::CommandFactory;
+    let source_id = clap::Id::from("Source");
+    let opt_command = Opt::command();
+    let source_group = opt_command
+        .get_groups()
+        .find(|g| g.get_id() == &source_id)
+        .unwrap();
+    assert!(source_group.is_required_set());
+    // assert!(source_group.is_multiple()); currently broken. Fixed by PR #4704
```

Comment:
Please make a note about this on the issue, not PR (as PRs come and go) and delete it.,

As-is, we have no way to track fixing this and it will get lost.